File: typescript-connector/src/models/Polygon.ts
// Class from DFC Business Ontology: #Polygon
import { SemanticObject } from "../core/SemanticObject.js";
import { Geometry, type GeometryParams } from "./Geometry.js";

export interface PolygonParams extends GeometryParams {}
export class Polygon extends Geometry {
  static get SEMANTIC_TYPE(): string {
    return "dfc-b:Polygon";
  }



  constructor(
    semanticId: string,
    params?: PolygonParams,
  ) {
    super(semanticId, params);
    this.semanticType = Polygon.SEMANTIC_TYPE;
  }
  static {
    SemanticObject.typeRegistry.set(Polygon.SEMANTIC_TYPE, Polygon);
  }
}

File: ruby-gem/lib/models/polygon.rb
# frozen_string_literal: true

# Class from DFC Business Ontology: #Polygon
require_relative '../core/semantic_object'
require_relative 'geometry'


module DfcLinkmlConnector
  module Models
    class Polygon < Geometry
      SEMANTIC_TYPE = "dfc-b:Polygon".freeze

      # @param semanticId [String]
      # @param coordinates: nil, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil
      def initialize(semanticId, coordinates: nil, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil)
        super(semanticId, coordinates: coordinates, date: date, description: description, name: name, characteristicOf: characteristicOf, dimension: dimension)

        self.semanticType = "dfc-b:Polygon"

      end

      Core::SemanticObject.type_registry[SEMANTIC_TYPE] = self
    end
  end
end
